python-package/lightgbm/engine.py

```diff
@@ -125,6 +128,7 @@ def train(
     """
     # create predictor first
     params = copy.deepcopy(params)
+    fobj = _objective_is_callable(params, fobj)
```

Comment:
I don't believe that adding a new internal function like this is necessary.

I think it would be sufficient, and simpler to add the following here

```python
params = _choose_param_value(
    main_param_name='objective',
    params=params,
    default_value=fobj
)
if callable(params["objective"]):
    fobj = params["objective"]
    params["objective"] = 'none'
```

And remove this code right below iit.

```python
if fobj is not None:
    for obj_alias in _ConfigAliases.get("objective"):
        params.pop(obj_alias, None)
    params['objective'] = 'none'
```

`_choose_param_value()`:

* handles resolving parameter aliases for `objective`
    - which, as of #4829, are sourced via the C API, to keep the Python package and C++ library consistent
* enforces the behavior "values in `params` should be preferred to keyword arguments"

@StrikerRUS what do you think?

Reply:
I like this proposal. The code will look much better I think.

Reply:
@jameslamb I like more that implementation is more clean and consistent and works the same